### PR TITLE
Register URL Model for Navigation

### PR DESCRIPTION
**Pull request by webbyfox**

### CODE REVIEW

File: djangocms_url_manager/models.py [hunk 1]
```diff
@@ -130,6 +130,9 @@ def get_url(self, site):
     def __str__(self):
         return self.get_url(self.site)
```
Comment (chaosk) [inline djangocms_url_manager/models.py]: Use get_absolute_url here then

File: djangocms_url_manager/cms_config.py [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+from .models import Url
+
 
 class UrlCMSAppConfig(CMSAppConfig):
     djangocms_url_manager_enabled = True
     url_manager_supported_models = [Page]
+    djangocms_navigation_enabled = getattr(
+        settings, "DJANGOCMS_NAVIGATION_CMS_MODELS_ENABLED", False
+    )
+    navigation_models = {Url: ["internal_name"]}
```
Comment (geoidesic) [inline djangocms_url_manager/cms_config.py]: You've added a field, but no explanation as to why?

File: djangocms_url_manager/forms.py [hunk 3]
```diff
@@ -121,124 +98,129 @@ def __init__(self, *args, **kwargs):
...
+        site = data.get("site")
 
         if url and url.site == site:
-            raise forms.ValidationError({
-                'site': _('Overriden site must be different from the original.'),  # noqa: E501
-            })
+            raise forms.ValidationError(
+                {
+                    "site": _(
+                        "Overriden site must be different from the original."
```
Comment (geoidesic) [inline djangocms_url_manager/forms.py]: I don't understand this message... I don't think a user would either.

File: djangocms_url_manager/cms_config.py (hunk 2 above)
Comment (webbyfox) [reply djangocms_url_manager/cms_config.py]: This is navigation functionality explain in the navigation app. Also, explain in the model.py level

File: djangocms_url_manager/forms.py (hunk 3 above)
Comment (webbyfox) [reply djangocms_url_manager/forms.py]: I formatted for lint nothing new added.

File: djangocms_url_manager/compat.py [hunk 4]
```diff
@@ -14,3 +14,9 @@ def get_page_placeholders(page, language=None):
         return page.get_placeholders()
     except TypeError:
         return page.get_placeholders(language)
+
+
+def is_navigation_enabled():
+    from djangocms_url_manager.cms_config import UrlCMSAppConfig
+
+    return UrlCMSAppConfig.djangocms_navigation_enabled
```
Comment (monikasulik) [inline djangocms_url_manager/compat.py]: Why is this done like this rather than via the django app registry? Although unlikely, it is possible that the class could have a different value than the instance (there's nothing stopping somebody doing some kind of conditional setting of this var in the `__init__` of the class for example). So we should be taking the instance that is in django's app registry and taking the value from there IMO.

File: djangocms_url_manager/models.py [hunk 5]
```diff
@@ -128,19 +137,31 @@ def get_url(self, site):
         return url
 
     def __str__(self):
+        return self.get_absolute_url()
+
+    def get_absolute_url(self):
```
Comment (monikasulik) [inline djangocms_url_manager/models.py]: I don't think `__str__` or `get_absolute_url` have tests?

File: djangocms_url_manager/compat.py [hunk 6]
```diff
@@ -14,3 +14,9 @@ def get_page_placeholders(page, language=None):
         return page.get_placeholders()
     except TypeError:
         return page.get_placeholders(language)
+
+
+def is_navigation_enabled():
```
Comment (monikasulik) [inline djangocms_url_manager/compat.py]: Also, why is this here if this is only used in tests?

File: tests/test_cms_config.py [hunk 7]
```diff
@@ -119,3 +124,57 @@ def test_url_manager_supported_tuple_model_without_function(self):
                 PollContent: get_all_poll_content_objects,
             }
         )
+
+
+@skipIf(not is_navigation_enabled(), "Test relevant only if navigation is enabled")
```
Comment (monikasulik) [inline tests/test_cms_config.py]: I'm not following why you have this skip here. Surely this depends on the test settings which are the same regardless of if we're running this on circle or on our local. What's more the tests below check are actually checking both the enabled and disabled case, right? So why would you only want to run them if it's enabled?

File: tests/test_cms_config.py [hunk 8]
```diff
@@ -119,3 +124,57 @@ def test_url_manager_supported_tuple_model_without_function(self):
...
+        self.app.cms_extension.navigation_apps_models = {}
+
+    def tearDown(self):
+        # Populate everything again so our setting changes do not effect
+        # any other tests
+        from djangocms_navigation import cms_config
+
+        self.app.cms_extension.navigation_apps_models = {}
+        self.app.cms_config = cms_config.NavigationCMSAppConfig(self.app)
+        configure_cms_apps([self.app])
```
Comment (monikasulik) [inline tests/test_cms_config.py]: AFAIK this way of repopulating causes test leaks.

File: tests/test_cms_config.py [hunk 9]
```diff
@@ -119,3 +124,57 @@ def test_url_manager_supported_tuple_model_without_function(self):
...
+
+    def test_cms_models_added_to_navigation_by_default(self):
+        from djangocms_navigation import cms_config
+
+        # The app should have a cms config with the overridden setting
+        self.app.cms_config = cms_config.NavigationCMSAppConfig(self.app)
+
+        configure_cms_apps([self.app])
+
+        self.assertIn(Page, self.app.cms_extension.navigation_apps_models)
```
Comment (monikasulik) [inline tests/test_cms_config.py]: Why `assertIn` not `assertListEqual` with an exact check of what's in there?

File: tests/test_cms_config.py [hunk 10]
```diff
@@ -119,3 +124,57 @@ def test_url_manager_supported_tuple_model_without_function(self):
...
+    @override_settings(DJANGOCMS_NAVIGATION_CMS_MODELS_ENABLED=True)
+    def test_cms_models_added_to_navigation_if_enabled(self):
+        from djangocms_navigation import cms_config
+
+        # The app should have a cms config with the overridden setting
+        self.app.cms_config = cms_config.NavigationCMSAppConfig(self.app)
+
+        configure_cms_apps([self.app])
+
+        self.assertIn(Page, self.app.cms_extension.navigation_apps_models)
```
Comment (monikasulik) [inline tests/test_cms_config.py]: ditto

File: tests/test_cms_config.py [hunk 11]
```diff
@@ -119,3 +124,57 @@ def test_url_manager_supported_tuple_model_without_function(self):
...
+    def test_cms_models_not_added_to_navigation_if_disabled(self):
+        from djangocms_navigation import cms_config
+
+        importlib.reload(cms_config)  # Reload so setting gets checked again
+        # The app should have a cms config with the overridden setting
+        self.app.cms_config = cms_config.NavigationCMSAppConfig(self.app)
+
+        configure_cms_apps([self.app])
+
+        self.assertNotIn(Page, self.app.cms_extension.navigation_apps_models)
```
Comment (monikasulik) [inline tests/test_cms_config.py]: Ditto, I suggest just checking it's an empty list.

File: djangocms_url_manager/models.py (hunk 5 above)
Comment (webbyfox) [reply djangocms_url_manager/models.py]: There are for str. I've added for get_absolute_url.
